modules/reindex/src/test/java/org/elasticsearch/reindex/AsyncBulkByScrollActionTests.java

```diff
@@ -826,6 +876,30 @@ public void done(TimeValue extraKeepAlive) {}
         expectThrows(IllegalArgumentException.class, () -> response.consumeHits(1));
     }
 
+    public void testEnableScrollByDefault() {
+        var preparedSearchRequest = AbstractAsyncBulkByScrollAction.prepareSearchRequest(testRequest, false, false);
+        assertThat(preparedSearchRequest.scroll(), notNullValue());
+    }
+
+    public void testDisableScrollWhenMaxDocsIsLessThenScrollSize() {
+        testRequest.setMaxDocs(1);
```

Comment:
Can we use random 1-100?

```suggestion
        testRequest.setMaxDocs(between(1,100));
```